### ISSUE TITLE
False positive with typechecker errors

### ISSUE THREAD

**Issue by skius** [JM4ier/program-verification-project-testrunner#1]
```bash
$ cat tests/required/04-sheet/sample.fail.spr
program { a := 2
return 2 }
```

correctly results in an error when encoding and verifying the result with Viper, in particular the following response is sent:

```json
{"msg_body":{"text":"Viper AST Constructor"},"msg_type":"copyright_report"}
{"msg_body":[],"msg_type":"warnings_during_parsing"}
{"msg_body":{"details":{"result":{"errors":[{"cached":false,"position":{"end":"5:15","file":"/workdir/build/viper/build/tests/required/04-sheet/sample.fail.vpr","start":"5:3"},"tag":"typechecker.error","text":"undeclared identifier `a`, expected parameter or local variable (sample.fail.vpr@5.3--5.15)"},{"cached":false,"position":{"end":"5:20","file":"/workdir/build/viper/build/tests/required/04-sheet/sample.fail.vpr","start":"5:19"},"tag":"typechecker.error","text":"found incompatible type `Int`, expected `<error>` (sample.fail.vpr@5.19--5.20)"}],"type":"error"},"time":419},"status":"failure"},"msg_type":"ast_construction_result"}
```

however, the test runner counts the program as if it verified, which means the test itself fails:

```
╭──────────────────────────────────────────────────────────────╮
│                           required                           │
sample.fail.spr                    pass=0    fail=[]
│                  required: passed 0/1 tests                  │
╰──────────────────────────────────────────────────────────────╯
{"msg":"shutting down..."}
╭──────────────────────────────────────────────────────────────╮
│                     finished test suite                      │
╰──────────────────────────────────────────────────────────────╯
```

(also, thanks for the runner!🚀)

**Comment by JM4ier** [comment]
should be fixed with df1917e - unfortunately I cannot confirm if it's truly fixed because my encoding doesn't *yet* cause errors for assigning to undeclared variables.

Can you check if it works as expected now?

**Comment by skius** [comment]
Yep, thanks!

```
╭──────────────────────────────────────────────────────────────╮
│                           required                           │
sample.fail.spr                    pass=0    fail=[]
│                  required: passed 1/1 tests                  │
╰──────────────────────────────────────────────────────────────╯
```
